src/color.rs

```diff
@@ -32,52 +32,84 @@ impl Display for Color {
+    /// <div style="background-color:rgb(0%, 50%, 0%); width: 10px; padding: 10px; border: 1px solid;"></div>
     pub const GREEN: Self = Self::new(0.0, 0.5, 0.0, 1.0);
-    /// <div style="background-color:new(0%, 100%, 0%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    /// <div style="background-color:rgb(0%, 100%, 0%); width: 10px; padding: 10px; border: 1px solid;"></div>
     pub const LIME: Self = Self::new(0.0, 1.0, 0.0, 1.0);
-    /// <div style="background-color:new(50%, 50%, 0%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    /// <div style="background-color:rgb(50%, 50%, 0%); width: 10px; padding: 10px; border: 1px solid;"></div>
     pub const OLIVE: Self = Self::new(0.5, 0.5, 0.0, 1.0);
-    /// <div style="background-color:new(100%, 100%, 0%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    /// <div style="background-color:rgb(100%, 100%, 0%); width: 10px; padding: 10px; border: 1px solid;"></div>
     pub const YELLOW: Self = Self::new(1.0, 1.0, 0.0, 1.0);
-    /// <div style="background-color:new(0%, 0%, 50%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    /// <div style="background-color:rgb(0%, 0%, 50%); width: 10px; padding: 10px; border: 1px solid;"></div>
     pub const NAVI: Self = Self::new(0.0, 0.0, 0.5, 1.0);
-    /// <div style="background-color:new(0%, 0%, 100%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    /// <div style="background-color:rgb(0%, 0%, 100%); width: 10px; padding: 10px; border: 1px solid;"></div>
     pub const BLUE: Self = Self::new(0.0, 0.0, 1.0, 1.0);
-    /// <div style="background-color:new(0%, 50%, 50%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    /// <div style="background-color:rgb(0%, 50%, 50%); width: 10px; padding: 10px; border: 1px solid;"></div>
     pub const TEAL: Self = Self::new(0.0, 0.5, 0.5, 1.0);
-    /// <div style="background-color:new(0%, 100%, 100%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    /// <div style="background-color:rgb(0%, 100%, 100%); width: 10px; padding: 10px; border: 1px solid;"></div>
     pub const AQUA: Self = Self::new(0.0, 1.0, 1.0, 1.0);
 
     /// Creates a Color from new components
     pub const fn new(r: f32, g: f32, b: f32, a: f32) -> Self {
         Self { r, g, b, a }
     }
 
-    /// Creates a Color from a hex value
-    pub fn from_hex(hex: u32) -> Self {
+    /// Creates a Color from a hex value.
+    /// The format of `hex` is `0xRRGGBB`, the MSB is discarded.
+    /// The `a` component is always set to 1.0.
+    pub fn from_hex_rgb(hex: u32) -> Self {
         let r = ((hex >> 16) & 0xFF) as f32 / 255.0;
         let g = ((hex >> 8) & 0xFF) as f32 / 255.0;
         let b = (hex & 0xFF) as f32 / 255.0;
         Self::new(r, g, b, 1.0)
     }
 
+    /// Creates a Color from a hex value.
+    /// The format of `hex` is `0xBBGGRR`, the MSB is discarded.
+    /// The `a` component is always set to 1.0.
+    pub fn from_hex_bgr(hex: u32) -> Self {
```

Comment:
When the hell would you ever want to use this? I think the selling point of gfx_maths is, that it is being simple. If you really want to have sth like this (I don't see a single use case though), I would implement it via macro.

Reply:
These functions cover most of the commonly used rgb pixel formats. E.g. some image formats store their pixels in BBGGRR instead of RRGGBB. I don't want to leave these common use cases unaccounted for.